src/cli/cluster/cluster_manager.js

```diff
@@ -31,7 +31,7 @@ process.env.kbnWorkerType = 'managr';
 export default class ClusterManager {
   static async create(opts = {}, settings = {}) {
     const transformedSettings = transformDeprecations(settings);
-    const config = await Config.withDefaultSchema(transformedSettings);
+    const config = Config.withDefaultSchema(transformedSettings);
```

Comment:
__note:__ I don't see any reason to make async that is really sync :) But the main reason for this change is that with `async` it's hard to reliably use `Config.withDefaultSchema` inside constructors.

Reply:
Looks like this was made `async` in https://github.com/elastic/kibana/pull/18926, maybe @kobelb can remember if we still need async config loading for Native Controllers for some reason?

Reply:
I seem to remember that it was a part of an initial attempt to support loading the package.json files from plugins without executing their code, but based on the current state of https://github.com/elastic/kibana/pull/18926 it doesn't seem necessary

Reply:
Interesting! Now I'm curious what was supposed to be `async` inside of `withDefaultSchema` @kobelb ? :)

Reply:
It's because of this: https://github.com/elastic/kibana/pull/17496/files#diff-346d12eb90facc1036ff15f93bd825bf

To be able to determine if the system call filters should be enabled by default, we have to manually check the OS version here, which is async: https://github.com/elastic/kibana/pull/17496/files#diff-12472d028fb5d43810d33bc6d77e93c4

Reply:
Is there any reason that this value can't be set within one of the `Native Controllers` mixins? Or [set during `serve` CLI command initialization phase](https://github.com/elastic/kibana/blob/master/src/cli/serve/serve.js#L101-L103)? Both these places support async operations.

Reply:
It could be set elsewhere, but using the default value of config schema itself seemed the right place to do so given that we'd like for it to default to what we determine but allow the user to override it's value.

Reply:
> It could be set elsewhere, but using the default value of config schema itself seemed the right place to do so

Yeah, I see your point.

> given that we'd like for it to default to what we determine but allow the user to override it's value.

That's also supported with the approaches I mentioned above (+ with this we'll call `getSystemCallFiltersEnabledDefault` only if it's not overridden via config file or CLI arg). I'm just not sure whether dynamic asynchronously constructed schema can bring more value than harm. In the new platform schema is defined as static property for every possible `***Config` class as well.

Anyway this PR is a blocker for Native Controllers, I'll keep your use case in mind and we'll figure out something together once https://github.com/elastic/kibana/pull/19994 is merged :)

Reply:
That makes sense, there have been a few other places where I've resorted to using the async default configs as well: https://github.com/elastic/kibana/blob/master/x-pack/plugins/reporting/index.js#L91 https://github.com/elastic/kibana/blob/master/x-pack/plugins/reporting/index.js#L98

Reply:
Hmm, thanks for sharing! You have quite a few smart defaults :)

Reply:
I somehow found myself in a position where I needed quite a few different defaults for different operating systems, with a really poor way of detecting support....

Reply:
One way we can handle this sort of thing in the new platform would be to construct an `Environment` object before other services that we can pass around to APIs like this. This way by the time something like this would need to determine the operating system, that value would have already been determined and made available synchronously through the `Environment`.

Reply:
We do exactly that (almost) with `env` and `installationFeatures` (`isClusterModeSupported`, `isOssModeSupported`, `isXPackInstalled` etc.) in new platform: https://github.com/elastic/kibana/pull/19994/files#diff-806634e0e6f42d8ea73cd3565c9089cbR31